recipes/json-schema-validator/all/conanfile.py

```diff
@@ -0,0 +1,86 @@
+import os
+import glob
+from conans import ConanFile, CMake, tools
+from conans.tools import Version
+from conans.errors import ConanInvalidConfiguration, ConanException
+
+class JsonSchemaValidatorConan(ConanFile):
+    name = "json-schema-validator"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/pboettch/json-schema-validator"
+    description = "JSON schema validator for JSON for Modern C++ "
+    topics = ("json-schema-validator", "modern-json",
+              "schema-validation", "json")
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt"]
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    short_paths = True
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        compiler_version = Version(self.settings.compiler.version)
+        if self.settings.os == "Windows"and self.settings.compiler == "Visual Studio" and compiler_version < "16":
+            try:
+                # Force check to ensure compatability
+                tools.check_min_cppstd(self, "17") # There's an issue in upstream forcing a higher cppstd version. See https://github.com/pboettch/json-schema-validator/issues/106
```

Comment:
```suggestion
                # There's an issue in upstream forcing a higher cppstd version. See https://github.com/pboettch/json-schema-validator/issues/106
                if self.settings.get_safe("cppstd"):
                    tools.check_min_cppstd(self, "17") 
```

Again, you can't check it without cppstd configured

Reply:
I tested it on both Windows and linux, it uses the compilers default for the comparison. which is exactly what I require here

Reply:
This is the behavoir [log](https://gist.github.com/prince-)chrismc/3004dfeda9992d109fdb4f40a7a82725

Reply:
> I tested it on both Windows and linux, it uses the compilers default for the comparison. which is exactly what I require here

You can't use the default compiler standard, because old compilers can support c++11 but use c++98 by default. Again, all official recipes check the cppstd, this is not new.

Reply:
I understand the rule, and I will follow it, it however forces higher compiler versions then necessary in cases like this